datafusion/spark/src/function/string/format_string.rs

```diff
@@ -0,0 +1,2374 @@
+            | ColumnarValue::Scalar(ScalarValue::Utf8View(Some(fmt))) => {
+                let formatter = Formatter::parse(fmt, &data_types)?;
+                let mut result = Vec::with_capacity(len.unwrap_or(1));
+                for i in 0..len.unwrap_or(1) {
+                    let scalars = args.args[1..]
+                        .iter()
+                        .map(|arg| try_to_scalar(arg.clone(), i))
+                        .collect::<Result<Vec<_>>>()?;
+                    let formatted = formatter.format(&scalars)?;
+                    result.push(formatted);
+                }
+                if is_scalar {
+                    match fmt_type {
+                        DataType::Utf8 => Ok(ColumnarValue::Scalar(ScalarValue::Utf8(
+                            Some(result.first().unwrap().clone()),
+                        ))),
+                        DataType::LargeUtf8 => Ok(ColumnarValue::Scalar(
+                            ScalarValue::LargeUtf8(Some(result.first().unwrap().clone())),
+                        )),
+                        DataType::Utf8View => Ok(ColumnarValue::Scalar(
+                            ScalarValue::Utf8View(Some(result.first().unwrap().clone())),
+                        )),
+                        _ => unreachable!(),
+                    }
+                } else {
+                    let array: ArrayRef = match fmt_type {
+                        DataType::Utf8 => Arc::new(StringArray::from(result)),
+                        DataType::LargeUtf8 => Arc::new(LargeStringArray::from(result)),
+                        DataType::Utf8View => Arc::new(StringViewArray::from(result)),
+                        _ => unreachable!(),
+                    };
+                    Ok(ColumnarValue::Array(array))
+                }
+            }
+            ColumnarValue::Array(fmts) => {
+                let mut result = Vec::with_capacity(len.unwrap());
+                for i in 0..len.unwrap() {
+                    let fmt = ScalarValue::try_from_array(fmts, i)?;
+                    match fmt.try_as_str() {
+                        Some(Some(fmt)) => {
+                            let formatter = Formatter::parse(fmt, &data_types)?;
+                            let scalars = args.args[1..]
+                                .iter()
+                                .map(|arg| try_to_scalar(arg.clone(), i))
+                                .collect::<Result<Vec<_>>>()?;
+                            let formatted = formatter.format(&scalars)?;
+                            result.push(Some(formatted));
+                        }
+                        Some(None) => {
+                            result.push(None);
+                        }
+                        _ => {
+                            return exec_err!(
+                                "Expected string type, got {:?}",
+                                fmt.data_type()
+                            )
+                        }
```

Comment:
Is reaching this arm possible? If I understand correctly, it's getting the strings out of the first array which should be guaranteed to be a type of string array by now, so it is a bit confusing to see this handling here